### PR TITLE
Fix config file loading in nbconvert.

### PR DESCRIPTION
**Pull request by Carreau**
Todo:

Failing test. 

ping @takluyver.

### CODE REVIEW

File: jupyter_core/application.py [hunk 1]
```diff
@@ -152,11 +152,8 @@ def load_config_file(self, suppress_errors=True):
         """
         self.log.debug("Searching %s for config files", self.config_file_paths)
         base_config = 'jupyter_config'
-        self.log.debug("Attempting to load config file: %s" %
```
Comment (minrk) [inline jupyter_core/application.py]: Don't remove this

File: jupyter_core/application.py [hunk 2]
```diff
@@ -164,15 +161,12 @@ def load_config_file(self, suppress_errors=True):
             # ignore errors loading parent
             self.log.debug("Config file %s not found", base_config)
             pass
-        
-        for config_file_name in self.config_files:
+        from itertools import chain
+        for config_file_name in chain(self.config_file_name, self.config_files):
             if not config_file_name or config_file_name == base_config:
                 continue
-            self.log.debug("Attempting to load config file: %s" %
```
Comment (minrk) [inline jupyter_core/application.py]: Don't remove this, either

File: jupyter_core/application.py (hunk 1 above)
Comment (Carreau) [inline jupyter_core/application.py]: It's moved inside Application.load_config_file now in traitlets.

Comment (Carreau) [inline jupyter_core/application.py]: Actually I need to pass log for it to work IIRC. Will do that,

Comment (minrk) [inline jupyter_core/application.py]: Ah, ok.

**Comment by Carreau** [comment]
Also added you both as collaborator here, in case you want to push on the branch directly.

**Comment by minrk** [comment]
Thanks, rebased to drop the merge-PR commit. Merging.

**Pull request merged**